### PR TITLE
feat: add update_receipt_rule to ses mock

### LINKED ISSUE #4489: `update_receipt_rule` for `ses` not implemented
It looks like `update_receipt_rule` is not implemented [here](https://github.com/spulec/moto/blob/master/IMPLEMENTATION_COVERAGE.md#ses), so I'll take a look into implementing this!

### PR DESCRIPTION
**Pull request by shermanhui**
This PR aims to resolve #4489

### CODE REVIEW

**Comment by shermanhui** [comment]
@bblommers do you mind adding the `hacktoberfest-accepted` label to this PR as well? 🙏🏻  I finished the challenge, but I'd like to keep track of the contributions I've made for this year! Thanks, let me know if anything else is needed to get this across the finish line - Happy Halloween! 👻  🎃

File: IMPLEMENTATION_COVERAGE.md [hunk 1]
```diff
@@ -1019,7 +1019,6 @@
 - [ ] bundle_instance
 - [ ] cancel_bundle_task
 - [ ] cancel_capacity_reservation
-- [ ] cancel_capacity_reservation_fleets
```
Comment (shermanhui) [inline IMPLEMENTATION_COVERAGE.md:1022]: @bblommers Is `make implementation_coverage` expected to remove these commands? I ran it in order to update the `ses` module coverage, but it looks like it goes through the entire repo and updates all the modules. I'm worried it's accidentally removing commands that are supposed to be kept. If that's the case, I can revert this commit and manually update the `ses` section.

Comment (bblommers) [reply IMPLEMENTATION_COVERAGE.md:1022]: It's not uncommon, to see methods removed. The script uses botocore in the background to check which methods AWS supports, and every now and then the AWS team will remove methods or an entire service completely. So nothing to worry about.

**Review by bblommers — APPROVED** [review]
LGTM! Thanks for the PR @shermanhui!

**Comment by bblommers** [comment]
> 
> 
> @bblommers do you mind adding the `hacktoberfest-accepted` label to this PR as well? 🙏🏻 I finished the challenge, but I'd like to keep track of the contributions I've made for this year! Thanks, let me know if anything else is needed to get this across the finish line - Happy Halloween! 👻 🎃

Done! Happy halloween 🎃 🙂

**Pull request merged**